src/communication/azext_communication/custom.py

```diff
@@ -1,16 +1,200 @@
+
+        if args.user_assigned:
+            user_assigned_flag = True
+            for identity in args.user_assigned:
+                user_dict[str(identity)] = {}
+
+        if user_assigned_flag:
+            user_dict.update(current_user_assigned_identities)
+            args.user_assigned_identities = user_dict
+
+        args.type = args_type_assignment(user_assigned_flag, system_assigned_flag)
+
+
+class IdentityRemove(_IdentityRemove):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        from azure.cli.core.aaz import AAZListArg, AAZResourceIdArg, AAZBoolArg, AAZResourceIdArgFormat
+        args_schema.system_assigned = AAZBoolArg(
+            options=["--system-assigned"],
+            help="""Enable system assigned identity.""",
+        )
+        args_schema.user_assigned = AAZListArg(
+            options=["--user-assigned"],
+            help="Space separated resource IDs to add user-assigned identities. Use [] to remove all identities.",
+        )
+
+        args_schema.user_assigned.Element = AAZResourceIdArg(
+            fmt=AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                                "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}"),
+        )
+        args_schema.type._registered = False
+        args_schema.type._required = False
+        args_schema.user_assigned_identities._registered = False
+        args_schema.user_assigned_identities._required = False
+        return args_schema
+
+    def pre_instance_update(self, instance):
+        from azure.cli.core.aaz import has_value
+        current_identity_type = ""
+        current_assigned_identities = {}
+        system_assigned_flag = False
+        user_assigned_flag = False
+        args = self.ctx.args
+        current_identity_type = instance.type
+        current_assigned_identities = instance.user_assigned_identities
+        current_assigned_identities = dict.fromkeys(current_assigned_identities.keys(), {})
+
+        del instance.user_assigned_identities
+
+        if "SystemAssigned" in str(current_identity_type):
+            system_assigned_flag = True
+        if "UserAssigned" in str(current_identity_type):
+            user_assigned_flag = True
+
+        user_dict = {}
+
+        if args.system_assigned:
+            system_assigned_flag = False
+
+        if has_value(args.user_assigned):
+            if len(args.user_assigned) == 0:
+                print("heeey")
+                current_assigned_identities = []
+            else:
+                for identity in args.user_assigned:
+                    print("neta")
+                    if str(identity) in current_assigned_identities:
+                        current_assigned_identities.pop(str(identity))
+                    else:
+                        raise CLIError(
```

Comment:
`CLIError` is not used any more. Please follow this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).